python/ray/train/v2/_internal/execution/controller/controller.py

```diff
@@ -348,22 +347,38 @@ def _step(self) -> TrainControllerLoopIterationResult:
             scaling_decision = (
                 self._scaling_policy.make_decision_for_non_running_worker_group()
             )
+            if isinstance(scaling_decision, NoopDecision):
+                next_state = InitializingState()
+            elif isinstance(scaling_decision, ResizeDecision):
+                next_state = SchedulingState(scaling_decision)
+            else:
+                raise ValueError(f"Unexpected scaling decision: {scaling_decision}")
+
             return TrainControllerLoopIterationResult(
                 run_attempt_id=self._get_run_attempt_id(),
                 previous_state=controller_state,
-                next_state=SchedulingState(scaling_decision),
+                next_state=next_state,
             )
         elif isinstance(controller_state, SchedulingState):
-            return self._execute_scaling_decision(controller_state.scaling_decision)
+            assert isinstance(controller_state.scaling_decision, ResizeDecision)
+            return self._execute_resize_decision(controller_state.scaling_decision)
         elif isinstance(controller_state, ReschedulingState):
             scaling_decision = (
                 self._scaling_policy.make_decision_for_non_running_worker_group()
             )
+            if isinstance(scaling_decision, NoopDecision):
+                next_state = ReschedulingState()
+            elif isinstance(scaling_decision, ResizeDecision):
+                next_state = SchedulingState(scaling_decision)
+            else:
+                raise ValueError(f"Unexpected scaling decision: {scaling_decision}")
```

Comment:
nit: these can all be a call to `_get_state_transition_from_scaling_decision(curr_state, decision)`